packages/kbn-grid-layout/grid/grid_panel/drag_handle/index.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useEffect, useImperativeHandle, useRef, useState } from 'react';
+
+import { GridLayoutStateManager, InteractionEventHandler } from '../../types';
+import { useGridLayoutEvents } from '../../use_grid_layout_events';
+import { DefaultDragHandle } from './default_drag_handle';
+
+export interface DragHandleApi {
+  setDragHandles: (refs: Array<HTMLElement | null>) => void;
+}
+
+export const DragHandle = React.forwardRef<
+  DragHandleApi,
+  {
+    gridLayoutStateManager: GridLayoutStateManager;
+    onInteractionEvent: InteractionEventHandler;
+    panelId: string;
+    rowIndex: number;
+  }
+>(({ onInteractionEvent, gridLayoutStateManager, panelId, rowIndex }, ref) => {
+  const attachLayoutEvents = useGridLayoutEvents({
+    onInteractionEvent,
+    interactionType: 'drag',
+    gridLayoutStateManager,
+    panelId,
+    rowIndex,
+  });
+
+  const [dragHandleCount, setDragHandleCount] = useState<number>(0);
+  const removeEventListenersRef = useRef<(() => void) | null>(null);
+
+  const setDragHandles = useCallback(
+    (dragHandles: Array<HTMLElement | null>) => {
+      setDragHandleCount(dragHandles.length);
+      for (const handle of dragHandles) {
+        if (handle === null) return;
+        handle.addEventListener('mousedown', attachLayoutEvents, { passive: true });
```

Comment:
we only have to attach the initiating the drag/resize events; the rest is attached on mouse down inside the `attachLayoutEvents`